letsencrypt/acme/messages.py

```diff
@@ -96,86 +60,55 @@ class Challenge(Message):
     :ivar list challenges: List of
         :class:`~letsencrypt.acme.challenges.Challenge` objects.
 
+    .. todo::
+        1. can challenges contain two challenges of the same type?
+        2. can challenges contain duplicates?
+        3. check "combinations" indices are in valid range
+        4. turn "combinations" elements into sets?
+        5. turn "combinations" into set?
+
```

Comment:
1. Yes
2. No, I don't believe there is a need for duplicate challenges, though it is supported technically in the spec.
4. 5. Both would be fine.